nuxeo-features/rest-api/nuxeo-rest-api-test/src/main/java/org/nuxeo/ecm/restapi/test/BaseTest.java

```diff
@@ -154,14 +155,31 @@ protected CloseableClientResponse getResponse(RequestType requestType, String pa
 
     protected CloseableClientResponse getResponse(RequestType requestType, String path, String data,
             MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers) {
+        return getResponse(requestType, path, data, queryParams, mp, headers, List.of());
+    }
+
+    protected CloseableClientResponse getResponse(RequestType requestType, String path, List<String> acceptedTypes) {
+        return getResponse(requestType, path, null, null, null, null, acceptedTypes);
+    }
+
+    protected CloseableClientResponse getResponse(RequestType requestType, String path, Map<String, String> headers,
+            List<String> acceptedTypes) {
+        return getResponse(requestType, path, null, null, null, headers, acceptedTypes);
+    }
+
+    protected CloseableClientResponse getResponse(RequestType requestType, String path, String data,
+            MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers,
+            List<String> acceptedTypes) {
```

Comment:
if i do that i will end with the issue below:

- there is already a method with the signature `protected CloseableClientResponse getResponse(RequestType requestType, String path)`  if i modify 

`CloseableClientResponse getResponse(RequestType requestType, String path, String data,MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers,` **to** `protected CloseableClientResponse getResponse(RequestType requestType, String path, String data, MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers,String... acceptedTypes)` it will works but that will means every time we need to call this method without fill in all parameters we should call it with passing `null` on all parameters there is no way to a add a shortcut method for example for my test [see](https://github.com/nuxeo/nuxeo/blob/594a88a3d35b9b963a7ebe57e5656a976bf6055f/nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/AuditTest.java#L95) will fails because it will call `protected CloseableClientResponse getResponse(RequestType requestType, String path)` and if i want to call the new one i  should `getResponse(BaseTest.RequestType.GET,
                "id/" + doc.getId() + "/@" + AuditAdapter.NAME, null, null, null, null, TEXT_CSV)`

i think the version with `List` it's better in this case that allow us to a shortcut method to avoid call it with many null parameters

`List.of()` it's no cost and i think most of the time they will be use the shortcut ones


@troger wdyt

Reply:
if i do that i will end with the issue below:

- there is already a method with the signature `protected CloseableClientResponse getResponse(RequestType requestType, String path)`  if i modify 

`getResponse(RequestType requestType, String path, String data,MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers,` 

**to** 

`getResponse(RequestType requestType, String path, String data, MultivaluedMap<String, String> queryParams, MultiPart mp, Map<String, String> headers,String... acceptedTypes)`

 it will works but that will means every time we need to call this method without fill in all parameters we should call it with passing `null` on all parameters there is no way to a add a shortcut method. 

For example for my test [see](https://github.com/nuxeo/nuxeo/blob/594a88a3d35b9b963a7ebe57e5656a976bf6055f/nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/AuditTest.java#L95) will fails because it will call `protected CloseableClientResponse getResponse(RequestType requestType, String path)` and if i want to call the new one i  should `getResponse(BaseTest.RequestType.GET,
                "id/" + doc.getId() + "/@" + AuditAdapter.NAME, null, null, null, null, TEXT_CSV)`

i think the version with `List` it's better in this case that allow us to a shortcut method to avoid call it with many null parameters

`List.of()` it's no cost and i think most of the time they will be use the shortcut ones


@troger wdyt